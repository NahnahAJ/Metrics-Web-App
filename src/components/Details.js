import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getOneApi } from '../redux/details';

const Details = () => {
  const dispatch = useDispatch();
  const artDetail = useSelector((state) => state.singleData.singleData);
  const { id } = useParams();
  useEffect(
    () => {
      dispatch(getOneApi(id));
    }, [dispatch, id],
  );

  return (

    <>
      <div>
        <p>{artDetail.buyPrice}</p>
        <h1>{artDetail.fileName}</h1>
        <h2>{artDetail.museumDesc}</h2>
        <img src={artDetail.imageUrl} alt={artDetail.image_id} />
      </div>
    </>
  );
};

export default Details;

// buyPrice,
// fileName,
// hasFake,
// id,
// imageUrl,
// museumDesc,
