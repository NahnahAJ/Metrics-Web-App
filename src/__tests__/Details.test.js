import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Details from '../components/Details';

const renderFuction = () => {
  render(
    <Provider store={store}>
      <Details />
    </Provider>,
  );
};

test('Check if details component Renders', () => {
  renderFuction();
  const linkElement = screen.getAllByRole('generic');
  expect(linkElement.length).toBeGreaterThan(1);
});
