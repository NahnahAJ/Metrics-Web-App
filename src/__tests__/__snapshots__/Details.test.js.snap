// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Details component snapshot 1`] = `
<DocumentFragment>
  <div
    class="detailsContainer"
  >
    <div
      class="imgContainer displays"
    >
      <img />
    </div>
    <table>
      <tbody>
        <tr>
          <td>
            Piece code
          </td>
          <td />
        </tr>
        <tr>
          <td>
            Name
          </td>
          <td />
        </tr>
        <tr>
          <td>
            Buying Price
          </td>
          <td>
            $
          </td>
        </tr>
        <tr>
          <td>
            Selling Price
          </td>
          <td>
            $
          </td>
        </tr>
        <tr>
          <td>
            Details
          </td>
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
